File: 01.jl
input = parse.(Int, readlines("01.txt"))

println(
    "Part 1: ", 
    count(i -> i > 0, diff(input))
)
println(
    "Part 2: ",
    count(i -> i > 0, diff([sum(input[i:i+2]) for i in 1:length(input) - 3]))
)

File: 01.R
# read & parse input
input <- as.numeric(readLines("01.txt", warn = FALSE))

# part 1
print(paste0(
  "Part 1: ", 
  sum(diff(input) > 0)
))

# part 2
print(paste0(
  "Part 2: ", 
  sum(diff(input, lag = 3) > 0)
))
